cmd/gapit/benchmark.go

```diff
@@ -585,34 +586,57 @@ func (verb *benchmarkVerb) Run(ctx context.Context, flags flag.FlagSet) error {
 
 	ctx = oldCtx
 	writeOutput := func() {
-		w := tabwriter.NewWriter(os.Stdout, 4, 4, 3, ' ', 0)
-		fmt.Fprintln(w, "Trace Time\tTrace Size\tTrace Frames\tState Serialization\tTrace Frame Time\tInteractive")
-		fmt.Fprintln(w, "----------\t----------\t------------\t-------------------\t----------------\t-----------")
-		fmt.Fprintf(w, "%v\t%v\t%v\t%v\t%v\t%v\n",
-			verb.traceDoneTime.Sub(verb.beforeStartTraceTime),
-			verb.traceSizeInBytes,
-			verb.traceFrames,
-			time.Duration(stateTime)*time.Nanosecond,
-			time.Duration(frameTime)*time.Nanosecond,
-			verb.gapisInteractiveTime.Sub(verb.traceDoneTime),
-		)
-		w.Flush()
 		preMecFramerate := float64(stateBuildStartTime - traceStartTimestamp)
 		if verb.StartFrame > 0 {
 			preMecFramerate = preMecFramerate / float64(verb.StartFrame)
 		}
-		fmt.Fprintln(os.Stdout, "")
-		w = tabwriter.NewWriter(os.Stdout, 4, 4, 3, ' ', 0)
-		fmt.Fprintln(w, "Caching Done\tInteraction\tMax Memory\tBefore MEC Frame Time\tTrailing Frame Time")
-		fmt.Fprintln(w, "------------\t-----------\t----------\t---------------------\t-----------------")
-		fmt.Fprintf(w, "%+v\t%+v\t%+v\t%+v\t%+v\n",
-			verb.gapisCachingDoneTime.Sub(verb.traceDoneTime),
-			verb.interactionDoneTime.Sub(verb.interactionStartTime),
-			traceMaxMemory,
-			time.Duration(preMecFramerate)*time.Nanosecond,
-			time.Duration(nonLoadingFrameTime)*time.Nanosecond,
-		)
-		w.Flush()
+		if verb.OutputCSV {
+			csvWriter := csv.NewWriter(os.Stdout)
+			defer csvWriter.Flush()
```

Comment:
Defer is a bit funny (and you probably don't actually want it here) but defer actually defers until the end of the current function (NOT the scope-block).
How I typically handle this is:
```
if verb.OutputCSV { 
   func() {
      csvWriter....
      defer csvwriter.Flush()
      ....
      ....
   } () 
 } else {
      ...
```

Reply:
Ah! Ok, didn't know that. Will change it and resubmit.

Reply:
It is super subtle, and goes completely against all of my past experience (thanks Go)